internal/jobmanager/job/manager.go

```diff
@@ -16,16 +16,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-var errNoMatchingBuild = errors.New("no matching build")
+var errNoMatchingJob = errors.New("no matching job")
 
 type jobManager struct {
```

Comment:
I would clearly separate the build and signing management features into two distinct interfaces. I get that the current kaniko job manager and the signing manager are doing similar tasks right now, but:
- upstream, we might move to Shipwright in the future, we means creating a new resource instead of a `Job`;
- downstream, we will work with either Shipwright or BuildConfig - no `Job` either.

Currently, the `Manager` interface in `build` abstracts any build implementation detail, which is convenient when we want to introduce a new build manager - we just provide it as is to the reconciler. The PR as is transforms it into a `Job` manager, which makes sense right now upstream, but might make things more challenging as we add more build systems.